### DISCUSSION TITLE
ACCESSORIES NOT FEEDBACK

### DISCUSSION THREAD

**Discussion by Le-Duc0069** [HomeSpan/HomeSpan#594] [Q&A]
my homekit has a problem like this, after a while the connection is successful, the accessory is not responding, i have to turn off wifi on iphone and turn it back on, is there any way to fix it homspan

**Comment by HomeSpan** [comment]
Have you tested this with an unmodified version of Example 1?  Are you using the latest version of iOS 16.5.1 with a Home Hub under the new architecture?

**Reply by Le-Duc0069** [reply]
I did as you said and it's still the same

**Comment by HomeSpan** [comment]
Can you provide more details.  What version of iOS are you using.  Are you using a Hub, such as a Home Pod?  Do you have any commercial devices that are causing the same problem?  Also, to fix the issue do you need to reboot the ESP32, or does simply turning WiFi on/off on the iPhone fixe the issue?  What if you instead close  and then re-open the Home App?  These types of connection issues plague the HomeKit ecosystem and unfortunately can be very hard to diagnose.  For example, some types of routers (certain mesh networks) can create problems for HomeKit (or, for example if you are running two wireless networks - one 5GHz and one 2.4GHz, and your router does not past traffic between the two).

**Reply by Le-Duc0069** [reply]
i'm using ios 16.5.1, don't use homehub, i have esp-32 on and off but still happens, only when i turn on/off wifi on iphone can it be controlled, i have combined 5GHz and 2.4 GHz becomes 1 network again

**Comment by HomeSpan** [comment]
I think the problem may be the lack of a home hub.   Apple released a new architecture for HomeKit earlier this year.   Have you updated to the new architecture (note this is independent of the latest versions of iOS)?  https://support.apple.com/en-us/HT213481

**Reply by Le-Duc0069** [reply]
So I have to buy a homehub to use it stably

**Reply by HomeSpan** [reply]
not necessarily - you may want to try updating to the new architecture first and see if it addresses the issue.  I'm not sure how well the new architecture works without a hub - Apple indicates that you need a hub to receive notifications, but you should be able to test with Example 1, since that does not use notifications.

**Reply by Le-Duc0069** [reply]
i updated to apple's new architecture but it's worse

**Reply by HomeSpan** [reply]
Sorry to hear it is still not working.  The symptoms you are describing are rather unique - when there are connection issues usually folks need to reboot the ESP32.  In your case you need to toggle WiFi on your phone, which suggests that the issue is not directly related to HomeSpan, but rather to how the iPhone is accessing the device.  It's unfortunately impossible to know whether adding a Home Hub will address the issue, so I'm hesitant to recommend buying a HomePod just to see if it fixes the problem.  However, if you happen to be buying one anyway for other reasons, you'll be able to check to see if it helps.

**Reply by Le-Duc0069** [reply]
I have tried to change another iphone and the result is the same
